cmd/minikube/cmd/start.go

```diff
@@ -447,6 +447,9 @@ func generateConfig(cmd *cobra.Command, k8sVersion string) (cfg.Config, error) {
 		if !cmd.Flags().Changed("docker-env") {
 			for _, k := range proxy.EnvVars {
 				if v := os.Getenv(k); v != "" {
+					// convert https_proxy to HTTPS_PROXY for linux
+					// TODO (@medyagh): if user has both http_proxy & HTTPS_PROXY set merge them.
```

Comment:
no idea on merge,  you mean just take one's precedence  over the other one ?

Reply:
ah ! thanks for correcting me, I meant that for no_proxy and NO_PROXY, if there are both merge them into one bigger NO_PROXY. good catch !

Reply:
ok, and also, would you mind me to REGISTRY_MIRROR env to EnvVars in the same file (proxy.go) ?

Reply:
i tried to use viper to load MINIKUBE_REGISTRY_MIRROR env

see PR https://github.com/kubernetes/minikube/pull/4607

Reply:
I am not familiar with  REGISTRY_MIRROR env var. the purpose of this PR is to plumb the standard networking env vars. (the ones that operating systems use to change their proxies). does REGISTRY_MIRROR env var change the behaviour of OS ? or is it minikube specific en var ?

if it is minikube specific that would be for a different PR

Reply:
on a second thought, is that env var a standard docker env var that everyone uses to set their registery mirror ?

Reply:
we also have another in progress PR by by @tstromberg  https://github.com/kubernetes/minikube/pull/3835/files

I think we can address the registry mirror in a separate PR

Reply:
@medyagh `REGISTRY_MIRROR` is  docker env var, it's important for users to pull images faster in China, no need to change here, i've send a PR, see https://github.com/kubernetes/minikube/pull/4607